aws/request/waiter_test.go

```diff
@@ -386,6 +386,217 @@ func TestWaiterError(t *testing.T) {
 	}
 }
 
+func TestWaiterRetryAnyError(t *testing.T) {
+	svc := &mockClient{Client: awstesting.NewClient(&aws.Config{
+		Region: aws.String("mock-region"),
+	})}
+	svc.Handlers.Send.Clear() // mock sending
+	svc.Handlers.Unmarshal.Clear()
+	svc.Handlers.UnmarshalMeta.Clear()
+	svc.Handlers.UnmarshalError.Clear()
+	svc.Handlers.ValidateResponse.Clear()
+
+	var reqNum int
+	results := []struct {
+		Out *MockOutput
+		Err error
+	}{
+		{ // retry
+			Err: awserr.New(
+				"MockException1", "mock exception message", nil,
+			),
+		},
+		{ // retry
+			Err: awserr.New(
+				"MockException2", "mock exception message", nil,
+			),
+		},
+		{ // success
+			Out: &MockOutput{
+				States: []*MockState{
+					{aws.String("running")},
+					{aws.String("running")},
+				},
+			},
+		},
+		{ // shouldn't happen
+			Out: &MockOutput{
+				States: []*MockState{
+					{aws.String("running")},
+					{aws.String("running")},
+				},
+			},
+		},
+	}
+
+	numBuiltReq := 0
+	svc.Handlers.Build.PushBack(func(r *request.Request) {
+		numBuiltReq++
+	})
+	svc.Handlers.Send.PushBack(func(r *request.Request) {
+		code := http.StatusOK
+		r.HTTPResponse = &http.Response{
+			StatusCode: code,
+			Status:     http.StatusText(code),
+			Body:       ioutil.NopCloser(bytes.NewReader([]byte{})),
+		}
+	})
+	svc.Handlers.Unmarshal.PushBack(func(r *request.Request) {
+		if reqNum >= len(results) {
+			t.Errorf("too many polling requests made")
+			return
+		}
+		r.Data = results[reqNum].Out
+		reqNum++
+	})
+	svc.Handlers.UnmarshalMeta.PushBack(func(r *request.Request) {
+		// If there was an error unmarshal error will be called instead of unmarshal
+		// need to increment count here also
+		if err := results[reqNum].Err; err != nil {
+			r.Error = err
+			reqNum++
+		}
+	})
+
+	w := request.Waiter{
+		MaxAttempts:      10,
+		Delay:            request.ConstantWaiterDelay(0),
+		SleepWithContext: aws.SleepWithContext,
+		Acceptors: []request.WaiterAcceptor{
+			{
+				State:    request.SuccessWaiterState,
+				Matcher:  request.PathAllWaiterMatch,
+				Argument: "States[].State",
+				Expected: "running",
+			},
+			{
+				State:    request.RetryWaiterState,
+				Matcher:  request.ErrorWaiterMatch,
+				Argument: "",
+				Expected: true,
```

Comment:
So this is the actual change on behavior, returning `true` when there's any error even if we don't have a matcher for it?

Reply:
No, this is an explicitly configured matcher saying "retry if there's any error".

Reply:
got it, where do we actually verify the new behavior then? We setup matchers, we verify we didn't get any error and that we only tried the expected amount of times, but I'm not sure where on these tests we check for the newly added boolean error

Reply:
`TestWaiterRetryAnyError` (this one) is configured to say "retry on any error" - so that tests the positive / `true` case - demonstrated by the first two mocked requests resulting in a retry.

Then `TestWaiterSuccessNoError` just says "immediately succeed on no error" - the negative / `false` case -  demonstrated by the very first mocked response terminating in success.